ebsco_adapter/ebsco_adapter/src/compare_uploads.py

```diff
@@ -0,0 +1,134 @@
+import os
+
+
+def find_uploads_to_compare(available_files, xml_s3_prefix, s3_store):
+    available_files_dates = list(available_files.keys())
+    dates_list = sorted(available_files_dates, reverse=True)
+
+    print(f"Available uploads: {dates_list}")
+
+    # Check if we've sent a notification for the date
+    notified_completion_flag = "notified.flag"
+    date_list_with_notified_flag = []
+    for date in dates_list:
+        notified_completion_flag_path = os.path.join(
+            xml_s3_prefix, date, notified_completion_flag
+        )
+        notified_completed = s3_store.file_exists(notified_completion_flag_path)
+        date_list_with_notified_flag.append(
+            {"date": date, "notified_completed": notified_completed}
+        )
+
+    # The current date is the most recent if it has not been notified
+    current_date = None
+    if (
+        len(date_list_with_notified_flag) > 0
+        and not date_list_with_notified_flag[0]["notified_completed"]
+    ):
+        current_date = date_list_with_notified_flag[0]["date"]
+
+    # The previous date is the next most recent if it has been notified
+    previous_date = None
+    if len(date_list_with_notified_flag) > 1:
+        previous_date = next(
+            (
+                date["date"]
+                for date in date_list_with_notified_flag[1:]
+                if date["notified_completed"]
+            ),
+            None,
+        )
+
+    comparison_results = {"previous": previous_date, "current": current_date}
+
+    print(
+        f"Comparing uploads from {comparison_results['previous']} to {comparison_results['current']}"
+    )
+
+    return comparison_results
+
+
+def find_updated_records(current_records, previous_records):
+    for id, record in current_records.items():
+        if id not in previous_records:
+            yield id, record
+        elif record["sha256"] != previous_records[id]["sha256"]:
+            yield id, record
+
+
+def find_deleted_records(current_records, previous_records):
+    for id, record in previous_records.items():
```

Comment:
This could be a generator expression, and we can use .keys() instead of .items, because we are ignoring the record at this point.

```suggestion
   return (id for id in previous.keys() if id not in current)
```